File: backend/app/schemas/security.py
from decimal import Decimal

from pydantic import BaseModel, ConfigDict


class SecuritySummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    ticker: str
    name: str
    exchange: str
    country: str
    sector: str | None = None
    asset_type: str
    current_price: Decimal | None = None
    market_cap_usd: int | None = None
    currency: str


class SecurityDetail(SecuritySummary):
    industry: str | None = None
    isin: str | None = None
    cusip: str | None = None
    sedol: str | None = None

File: web/src/types/security.ts
export interface SecuritySummary {
  id: number;
  ticker: string;
  name: string;
  exchange: string;
  country: string;
  sector: string | null;
  asset_type: string;
  current_price: number | null;
  market_cap_usd: number | null;
  currency: string;
}

export interface SecurityDetail extends SecuritySummary {
  industry: string | null;
  isin: string | null;
  cusip: string | null;
  sedol: string | null;
}
